File: app/java/src/main/java/com/example/production/domain/model/plan/MasterProductionSchedule.java
package com.example.production.domain.model.plan;

import lombok.*;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class MasterProductionSchedule {
    private Integer id;
    private String mpsNumber;
    private LocalDate planDate;
    private String itemCode;
    private BigDecimal planQuantity;
    private LocalDate dueDate;
    @Builder.Default
    private PlanStatus status = PlanStatus.DRAFT;
    private String locationCode;
    private String remarks;
    private LocalDateTime createdAt;
    private String createdBy;
    private LocalDateTime updatedAt;
    private String updatedBy;
}

File: app/csharp/src/ProductionManagement.Domain/Models/Plan/MasterProductionSchedule.cs
namespace ProductionManagement.Domain.Models.Plan;

/// <summary>
/// 基準生産計画（MPS）
/// </summary>
public class MasterProductionSchedule
{
    private string _statusValue = "草案";

    public int Id { get; set; }

    public required string MpsNumber { get; set; }

    public DateOnly PlanDate { get; set; }

    public required string ItemCode { get; set; }

    public decimal PlanQuantity { get; set; }

    public DateOnly DueDate { get; set; }

    /// <summary>
    /// ステータス（データベース格納用の日本語文字列）
    /// </summary>
    public string StatusValue
    {
        get => _statusValue;
        set => _statusValue = value;
    }

    /// <summary>
    /// ステータス（ドメインロジック用の Enum）
    /// </summary>
    public PlanStatus Status
    {
        get => PlanStatusExtensions.FromDisplayName(_statusValue);
        set => _statusValue = value.GetDisplayName();
    }

    public string? LocationCode { get; set; }

    public string? Remarks { get; set; }

    public DateTime CreatedAt { get; set; }

    public string? CreatedBy { get; set; }

    public DateTime UpdatedAt { get; set; }

    public string? UpdatedBy { get; set; }

    // リレーション
    public Item.Item? Item { get; set; }
}
